src/main/java/uk/gov/hmcts/divorce/divorcecase/model/CaseDocuments.java

```diff
@@ -199,4 +200,12 @@ public Optional<ListValue<DivorceDocument>> getDocumentGeneratedWithType(final D
             .filter(document -> documentType.equals(document.getValue().getDocumentType())).findFirst()
             : Optional.empty();
     }
+
+    @JsonIgnore
+    public void removeDocumentGeneratedWithType(final DocumentType documentType) {
+        if (!isEmpty(this.getDocumentsGenerated())) {
+            this.getDocumentsGenerated()
+                .removeIf(document -> CONDITIONAL_ORDER_GRANTED.equals(document.getValue().getDocumentType()));
```

Comment:
Is the documentType param supposed to be getting used instead of `CONDITIONAL_ORDER_GRANTED `?

Reply:
yes, thanks for spotting that